src/components/checkbox/checkbox.js

```diff
@@ -97,6 +97,19 @@ function MdCheckboxDirective(inputDirective, $mdAria, $mdConstant, $mdTheming, $
       var containerCtrl = ctrls[0];
       var ngModelCtrl = ctrls[1] || $mdUtil.fakeNgModel();
       var formCtrl = ctrls[2];
+      var labelHasLink = element.find('a').length > 0;
+
+      if (labelHasLink) {
+        var labelId = 'label-' + $mdUtil.nextUid();
+        attr.$set('aria-labelledby', labelId);
+
+        var label = element.children()[1];
+        var labelContents = label.innerHTML;
+        label.remove();
+        element.after('<span id="' + labelId + '" class="md-checkbox-link-label">' + labelContents + '</span>');
```

Comment:
This looks like an XSS vector if the label has any user-provided content. Can you do this without doing the HTML concatenation?